Migrate project from NAN to Node-API for improved compatibility

SanjayUG opened the pull request:
Hello there, This pr is the contribution to issue #1036,

Here's a summary of the changes made to migrate the project from nan to Node-API:
_src/addon.cpp_
Replaced #include <nan.h> with #include <node_api.h>.
Updated the IsRunningInElectronRenderer function to use Node-API functions instead of nan functions.
Modified the InitModule function to use Node-API for module initialization.

_src/shadow_node.hpp_
Replaced #include <nan.h> with #include <node_api.h>.
Changed the class inheritance from Nan::ObjectWrap to Napi::ObjectWrap<ShadowNode>.
Updated method signatures to use Napi::CallbackInfo instead of Nan::FunctionCallbackInfo<v8::Value>.
Replaced Nan::Persistent<v8::Function> with Napi::FunctionReference.

_src/rcl_handle.cpp_
Replaced #include <nan.h> with #include <node_api.h>.
Changed the class inheritance from Nan::ObjectWrap to Napi::ObjectWrap<RclHandle>.
Updated method signatures to use Napi::CallbackInfo instead of Nan::FunctionCallbackInfo<v8::Value>.
Replaced Nan functions with their Node-API equivalents in the Init, New, SyncProperties, PropertiesGetter, Release, Dismiss, and NewInstance methods.

_src/rcl_bindings.cpp_
Replaced #include <nan.h> with #include <node_api.h>.
Updated the Init and CreateNode methods to use Node-API functions instead of Nan functions.
src/rcl_action_bindings.cpp
Updated the ActionCreateServer method to use Node-API instead of nan.

If any feedback feel free to mention.

SanjayUG committed:
Migrate project from NAN to Node-API for improved compatibility

SanjayUG committed:
Refactor: Migrate NAN methods to Node-API in rcl_action_bindings.cpp and fix binding.gyp syntax error

minggangw commented:
Just wondering if we could leverage AI agent to convert most of the code from NAPI to Node-API :smile:

SanjayUG commented:
> Just wondering if we could leverage AI agent to convert most of the code from NAPI to Node-API 😄


Glad for your reply, 
I believe using AI in efficient way will help in the problem solving faster.

SanjayUG committed:
feat: Convert N-API code to Node-API

SanjayUG commented:
Can you just check it now,

I just modified the following things,
- Header File: Included node_api.h in both files.
- Initialization Function: Updated the initialization function to use Node-API conventions.
- Module Registration: Used NAPI_MODULE macro for module registration.

 i welcome you with some feedback to it.

minggangw commented on binding.gyp:
```diff
@@ -28,8 +28,7 @@
       'include_dirs': [
-        '.',
-        "<!(node -e \"require('nan')\")",
+        '<!@(node -p "require(\"node-addon-api\").include")',
```
I notice that a consistent failure on Actions https://github.com/RobotWebTools/rclnodejs/actions/runs/13857494263/job/38777515813?pr=1088#step:7:25, can you pass the compiling locally?
